src/application/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,211 @@
+// Importing packages
+import { CruxAssetTranslator, IPutAddressMapFailures, IPutAddressMapSuccess, IResolvedClientAssetMap } from "../../application/services/crux-asset-translator";
+import { CruxDomain } from "../../core/entities/crux-domain";
+import { CruxSpec } from "../../core/entities/crux-spec";
+import { CruxUser, IAddress, IAddressMapping } from "../../core/entities/crux-user";
+import { ICruxBlockstackInfrastructure } from "../../core/interfaces";
+import { ICruxUserRepository } from "../../core/interfaces/crux-user-repository";
+import { IKeyManager } from "../../core/interfaces/key-manager";
+import { ICruxIDState, setCacheStorage } from "../../index";
+import { BasicKeyManager } from "../../infrastructure/implementations/basic-key-manager";
+import { BlockstackCruxDomainRepository } from "../../infrastructure/implementations/blockstack-crux-domain-repository";
+import { BlockstackCruxUserRepository } from "../../infrastructure/implementations/blockstack-crux-user-repository";
+import { BaseError, CruxClientError, ErrorHelper, PackageErrorCode } from "../../packages/error";
+import { CruxDomainId, CruxId, InputIDComponents } from "../../packages/identity-utils";
+import { InMemStorage } from "../../packages/inmem-storage";
+import { StorageService } from "../../packages/storage";
+import { cloneValue } from "../../packages/utils";
+
+export const throwCruxClientError = (target: any, prop: any, descriptor?: { value?: any; }): any => {
+    let fn: any;
+    let patchedFn: any;
+    if (descriptor) {
+        fn = descriptor.value;
+    }
+    return {
+        configurable: true,
+        enumerable: true,
+        get() {
+            if (!patchedFn) {
+                patchedFn = async (...params: any[]) => {
+                    try {
+                        return await fn.call(this, ...params);
+                    } catch (error) {
+                        throw CruxClientError.fromError(error);
+                    }
+                };
+            }
+            return patchedFn;
+        },
+        set(newFn: any) {
+            patchedFn = undefined;
+            fn = newFn;
+        },
+    };
+};
+export interface ICruxWalletClientOptions {
+    privateKey?: string;
```

Comment:
this can be doubled as keyManager input also right?

Reply:
Fixed and updated